compute/clusters.go

```diff
@@ -290,59 +290,54 @@ func (a ClustersAPI) GetOrCreateRunningCluster(name string, custom ...Cluster) (
 	return a.Create(r)
 }
 
-// GetSmallestNodeTypeWithStorage returns smallest runtime node type
-func (a ClustersAPI) GetSmallestNodeTypeWithStorage() string {
-	nodeTypes, err := a.ListNodeTypes()
-	if err != nil || len(nodeTypes) == 0 {
+// NodeTypeRequest is a wrapper for local filtering of node types
+type NodeTypeRequest struct {
+	MinMemoryGB int32  `json:"min_memory_gb,omitempty"`
+	GBPerCore   int32  `json:"gb_per_core,omitempty"`
+	MinCores    int32  `json:"min_cores,omitempty"`
+	MinGPUs     int32  `json:"min_gpus,omitempty"`
+	LocalDisk   bool   `json:"local_disk,omitempty"`
+	Category    string `json:"category,omitempty"`
+}
+
+// GetSmallestNodeType returns smallest (or default) node type id given the criteria
+func (a ClustersAPI) GetSmallestNodeType(r NodeTypeRequest) string {
+	list, _ := a.ListNodeTypes()
+	// error is explicitly ingored here, because Azure returns
+	// apparently too big of a JSON for Go to parse
+	if len(list.NodeTypes) == 0 {
 		if a.client.IsAzure() {
 			return "Standard_D3_v2"
 		}
 		return "i3.xlarge"
 	}
```

Comment:
I'm not sure that it's a correct behavior to return default value in case if people specified some requirements that need to be fulfilled...
maybe we need to look to the other libraries that can parse large JSONs? I've seen references to https://github.com/tamerh/jsparser, for example

Reply:
i've documented it as a known issue -
https://github.com/databrickslabs/terraform-provider-databricks/blob/fabfcdda485050e444e97a74809a1a669a7000bf/docs/data-sources/node_type.md - does it seem reasonable? :) 

https://github.com/tamerh/jsparser is immature.

Reply:
that's ok - I went top-down, leaving comments, so the documentation was aftewards...